src/plugins/event-plugins/TrackerPlugin.ts

```diff
@@ -0,0 +1,209 @@
+import { Plugin, PluginContext } from '../Plugin';
+import { MonkeyPatch, MonkeyPatched } from '../MonkeyPatched';
+
+type Send = () => void;
+type Open = (method: string, url: string, async: boolean) => void;
+type XhrDetails = {
+    method: string;
+    url: string;
+    async: boolean;
+};
+type Fetch = typeof fetch;
+
+export const TRACKER_PLUGIN_ID = 'com.amazonaws.rum.tracker';
+
+/**
+ * TrackerPlugin is responsible for intercepting outgoing XMLHttpRequests and Fetch requests
+ * that are required to calculate the page load timing for single page applications (SPA).
+ * The plugin adds an eventListener to individual requests listening for "loadend",
+ * which is an indication of the request completing, regardless of success or failure.
+ *
+ * XMLHttpRequests have two phases
+ * (1) Open: this is where the configurations for the request is defined.
+ * (2) Send: this is when the actual network activity happens.
+ *
+ * During the send, the plugin will save the request into one of two locations:
+ * (1) If the current Page is not loaded, the request will be added to the Page object's
+ * ongoingActivity set, since we are certain the request belongs to the current Page.
+ * (2) If the current Page is loaded, we are not sure whether a route change will happen soon,
+ * or this is a request for some other purpose. As a result, we store this request into
+ * the PageManager's requestCache. If route change is detected, we can assume this request
+ * belongs to the route change, so we move the requests in the cache into the newly created
+ * Page object's ongoingActivity.
+ *
+ * Once the request is completed, it will mark the time of completion to calculate the current
+ * Page object's latestEndTime (if in ongoingActivity and page has not loaded),
+ * or simply removed from requestCache (if page has loaded).
+ *
+ * For Fetch, we use the input url combined with the init? parameter.
+ * Intuition behind this is that requests that have the same URL will most likely be
+ * POST requests, that have the init? parameter populated. As a result, combining init? with
+ * input url will ensure trackerPlugin can distinguish different fetch requests with same URLs.
```

Comment:
suggestion: Consider the audience when creating 'javadoc' documentation vs code comments. A javadoc (or in this case jsdoc) reader is likely interested in the what the class provides (timing virtual page loads), while a code comment reader is likely interested in implementation details. 

Also see the [Google Java Style Guide: The Summary Fragment](https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment).

Reply:
Thanks for the suggestion, duly noted for future iterations!